Core/Functions/String.lua

```diff
@@ -181,33 +181,89 @@ function F.String.WALink(class)
   return TXUI.IsClassic and I.Strings.WALinks.DEFAULT_CLASSIC or I.Strings.WALinks.DEFAULT
 end
 
+function F.String.RemoveRuneOfThePrefix(text)
+  return text:gsub(".* the ", ""):gsub(".* of ", "")
+end
+
+function F.String.RemoveEveryOfTheAndEverythingAfter(text)
+  return text:gsub(" of (.*)", ""):gsub(" the (.*)", "")
+end
+
+-- Capture the following strings
+-- .+%s(.+)$
+-- . represents all characters
+-- + makes us capture 1 or more repetitions of the previous character/symbol will always match the longest possible part
+-- %s represents all space characters
+-- $ at the end makes the pattern match to the end of the string
+-- Example: "Lightweave Embroidery" captures "Embroidery"
+function F.String.GetTheLastWordOfAString(text)
+  return strmatch(text, ".+%s(.+)$")
+end
+
+-- Capture the following string
+-- ^[%s%p]*
+-- ^ forces us to start capturing at the start of the string
+-- %s represents all space characters
+-- %p represents all punctuation characters
+-- * matches 0 or more repetitions of the previous character/symbol/pattern
+-- [] is a capture group
+-- This would capture the start of the string and replace all spaces with nothing (but most likely isn't working)
+function F.String.RemoveAllWhitespaceCharacters(text)
+  return text:gsub("^[%s%p]*", "")
+end
+
+-- Capture the following string
+-- %d+
+-- %d represents all digits
+-- this would capture the longest number chain in a string
+function F.String.ContainsNumericalCharacters(text)
```

Comment:
I feel like this is conflicting. Coment says "captures the longest number chain in a string" so I assume from `foo bar 123 4567` it would return `4567`? If so, `ContainsNumericalCharacters` doesn't actually reflect what the function does. If we go by the comment, I'd call this something like `GetLongestNumber` or something

Reply:
Hmm now I see we do have `GetTheNumberChain` so maybe the comment could be edited to clarify my previous comment?

Reply:
It is a different return value, which is why it's different. `ContainsNumericalCharacters` returns true or false due to using strmatch while `GetTheNumberChain` actually gets the value. But I can adjust this no prob.